lib/matrix_check_sum.rb

```diff
@@ -3,8 +3,40 @@
 # of numbers in row i is the same as the sum of numbers in column i for i = 0 to row.length-1
 # If this is the case, return true. Otherwise, return false.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) - n is the number of elements in the matrix
+# Space complexity: O(n) - n is the number of arrays in the matrix
```

Comment:
You have solved this with O(n) space complexity.  However it could be done with O(1) space complexity.  Think about how you could sum up a row and it's corresponding column together.